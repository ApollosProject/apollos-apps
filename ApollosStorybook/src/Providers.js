import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
