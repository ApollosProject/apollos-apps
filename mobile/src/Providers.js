import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
