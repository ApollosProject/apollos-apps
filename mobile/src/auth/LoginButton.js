import React, { PureComponent } from 'react';
import { withNavigation } from 'react-navigation';
import gql from 'graphql-tag';
import { Query, Mutation } from 'react-apollo';
import PropTypes from 'prop-types';

import { ButtonLink } from '/mobile/ui/Button';
import styled from '/mobile/ui/styled';
import ActivityIndicator from '/mobile/ui/ActivityIndicator';

import logout from './logout.graphql';

const Button = styled(({ theme }) => ({
  paddingHorizontal: theme.sizing.baseUnit,
}))(ButtonLink);

export const getLoginState = gql`
  query {
    isLoggedIn: authToken @client
  }
`;

@withNavigation
class LoginButton extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      push: PropTypes.func,
    }),
  };

  handleLoginPress = () => this.props.navigation.push('Auth');

  renderLogoutButton = () => (
    <Mutation mutation={logout}>
      {(handleLogout) => <Button onPress={() => handleLogout()}>Logout</Button>}
    </Mutation>
  );

  render() {
    return (
      <Query query={getLoginState}>
        {({ data: { isLoggedIn = false, loading } }) => {
          if (loading) return <ActivityIndicator />;
          if (isLoggedIn) return this.renderLogoutButton();
          return <Button onPress={this.handleLoginPress}>Login</Button>;
        }}
      </Query>
    );
  }
}

export default LoginButton;
