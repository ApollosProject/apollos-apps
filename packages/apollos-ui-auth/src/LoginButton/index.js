import React from 'react';
import { Query } from 'react-apollo';

import { Button } from '@apollosproject/ui-kit';

import { GET_LOGIN_STATE } from '../queries';
import { AuthConsumer } from '../Provider';

const LoginButton = (props) => (
  <Query query={GET_LOGIN_STATE}>
    {({ data }) => {
      const { isLoggedIn, loading } = data;
      if (isLoggedIn) return null;
      return (
        <AuthConsumer>
          {({ navigateToAuth }) => (
            <Button
              onPress={navigateToAuth}
              title="Get Connected"
              loading={loading}
              {...props}
            />
          )}
        </AuthConsumer>
      );
    }}
  </Query>
);

export default LoginButton;
