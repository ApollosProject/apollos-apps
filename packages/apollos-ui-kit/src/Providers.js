import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
