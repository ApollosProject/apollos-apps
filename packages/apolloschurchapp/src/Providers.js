import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
