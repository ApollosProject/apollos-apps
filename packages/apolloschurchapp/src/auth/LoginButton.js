import React, { PureComponent } from 'react';
import { withNavigation } from 'react-navigation';
import { Query, Mutation } from 'react-apollo';
import PropTypes from 'prop-types';

import { ButtonLink } from 'apolloschurchapp/src/ui/Button';
import styled from 'apolloschurchapp/src/ui/styled';
import ActivityIndicator from 'apolloschurchapp/src/ui/ActivityIndicator';

import logout from './logout';
import getLoginState from './getLoginState';

const Button = styled(({ theme }) => ({
  paddingHorizontal: theme.sizing.baseUnit,
}))(ButtonLink);

class LoginButton extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      push: PropTypes.func,
    }),
  };

  handleLoginPress = () => this.props.navigation.push('Auth');

  renderLogoutButton = () => (
    <Mutation mutation={logout}>
      {(handleLogout) => <Button onPress={() => handleLogout()}>Logout</Button>}
    </Mutation>
  );

  render() {
    return (
      <Query query={getLoginState}>
        {({ data: { isLoggedIn = false, loading } }) => {
          if (loading) return <ActivityIndicator />;
          if (isLoggedIn) return this.renderLogoutButton();
          return <Button onPress={this.handleLoginPress}>Login</Button>;
        }}
      </Query>
    );
  }
}

export default withNavigation(LoginButton);
