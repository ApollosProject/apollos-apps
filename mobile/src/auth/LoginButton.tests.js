import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from '/mobile/Providers';

import LoginButton, { getLoginState } from './LoginButton';

describe('LoginButton component', () => {
  it('renders a LoginButton when logged out', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: null },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LoginButton navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });

  it('renders a LogoutButton when logged in', async () => {
    const mock = {
      request: {
        query: getLoginState,
      },
      result: {
        data: { isLoggedIn: 'some-auth-token' },
      },
    };
    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <LoginButton navigation={navigation} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
