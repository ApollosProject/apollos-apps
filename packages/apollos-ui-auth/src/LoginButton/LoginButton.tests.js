import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import { Providers } from '@apollosproject/ui-kit';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { MockedProvider } from 'react-apollo/test-utils';
import AuthProvider, { GET_AUTH_TOKEN } from '../Provider';

import { GET_LOGIN_STATE } from '../queries';
import LoginButton from '.';

const cache = new InMemoryCache();
describe('LoginButton component', () => {
  it('renders nothing when logged in', async () => {
    await cache.writeQuery({
      query: GET_AUTH_TOKEN,
      data: { authToken: 'test' },
    });

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <MockedProvider cache={cache}>
        <AuthProvider>
          <Providers>
            <LoginButton navigation={navigation} />
          </Providers>
        </AuthProvider>
      </MockedProvider>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });

  it('renders a LoginButton when logged out', async () => {
    const mock = {
      request: {
        query: GET_LOGIN_STATE,
      },
      result: {
        data: { isLoggedIn: null },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <MockedProvider mocks={[mock]}>
        <Providers>
          <LoginButton navigation={navigation} />
        </Providers>
      </MockedProvider>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });

  it('renders a LoginButton that is loading', async () => {
    const mock = {
      request: {
        query: GET_LOGIN_STATE,
      },
      result: {
        data: { isLoggedIn: null },
      },
    };

    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(
      <MockedProvider mocks={[mock]}>
        <Providers>
          <LoginButton navigation={navigation} loading />
        </Providers>
      </MockedProvider>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
