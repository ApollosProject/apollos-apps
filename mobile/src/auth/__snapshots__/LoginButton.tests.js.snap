// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton component renders a LoginButton when logged out 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "color": "#17B582",
      "paddingHorizontal": 20,
    }
  }
>
  Login
</Text>
`;

exports[`LoginButton component renders a LogoutButton when logged in 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "color": "#17B582",
      "paddingHorizontal": 20,
    }
  }
>
  Logout
</Text>
`;
